### ISSUE TITLE
Placement of newly opened Component Browser when nodes are selected

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#4266]
This task is automatically imported from the old Task Issue Board and it was originally created by Adam Obuchowicz.
Original issue is [here](https://www.pivotaltracker.com/story/show/180887079).
<hr />

## Why
**As** Enso IDE user 
**I want** to have searcher placed under the selected nodes 
**So that** my graph during work will be nicely laid out

## Notes
Applying the Magnet algorithm is not in scope of this task.

## Acceptance Criteria

```gherkin
Scenario: 
GIVEN I opened the default newly-created project,
 and have bottom node selected
WHEN I open the Node Searcher with tab
OR WHEN I open the Node SEarcher with the (+) button
THEN the Node Searcher input is under the selected node and left-aligned to it
WHEN I accept some suggestion creating new node, restore selection 
  as it was, and add open the Node Searcher with tab
THEN the searcher input is vertically aligned to the previously created node and is placed directly to the left of it

GIVEN I opened the default newly-created project, move the top node to the left and select both nodes
WHEN I open the Node Searcher with tab
OR WHEN I open the Node Searcher with the (+) button
THEN  the Node Searcher input is left-aligned to the top node and is directly below the second node
WHEN I accept some suggestion creating new node, restore selection 
  as it was, and add open the Node Searcher with tab
THEN the searcher input is vertically aligned to the previously created node and is placed directly to the left of it
```

## Implementation Notes

All the logic should be implemented in the Graph Editor component: make a method calculating the position of a new node based on selection; the selection may be obtained from the `ide_view_graph_editor::Nodes` structure. Then the method shall be used in Graph Editor's FRP network.

## Blockers:
[#180887253](https://www.pivotaltracker.com/story/show/#180887253) resolved

## Comments:
The description of node placement is not clear to me. I think we should strictly use the words `left-aligned`, `vertically/horizontally aligned`.  

So for scenario 1: `the Node Searcher input is under the selected node and left-aligned to it` and `the searcher input is vertically aligned to the previously created node and is placed directly to the left of it`.
For scenario 2: `the Node Searcher input is left-aligned to the top node and is directly below the second node` and `the searcher input is vertically aligned to the previously created node and is placed directly to the left of it`.

By "directly" I mean "by the standard gap distance". (Ilya Bogdanov - Jan 31, 2022)
<hr />
Suggestions applied (Adam Obuchowicz - Jan 31, 2022)
<hr />
**Technical Design** review set to **pass** (Adam Obuchowicz - Jan 31, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-03-23):

**Progress:** Implemented and tested a quick and dirty prototype of the feature. Found one design question (planning to ask & resolve tomorrow). It should be finished by 2022-03-26. (Enso Bot - Mar 24, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-03-24):

**Progress:** After a chat with WD, modified the code so that when a visualization is visible & enabled, new nodes are positioned below the visualization (they were positioned to the left of a visualization previously). In other words, new nodes will now be positioned below the bounding box of a node, instead of below the input area of a node. Refactored the initial PoC code to a more elegant design. Opened a draft PR and recorded a screencast of the feature. It should be finished by 2022-03-26. (Enso Bot - Mar 25, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for today (2022-03-25):

**Summary:** There is **2 days** delay in implementation of the [Placement of newly opened Component Browser when nodes are selected (#180887079) task](https://www.pivotaltracker.com/story/show/180887079).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Forgot to account for the weekend when estimating the End Date last time. (Enso Bot - Mar 25, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for today (2022-03-25):

**Progress:** Did some polishing and refactoring for a PR. Opened a PR. It should be finished by 2022-03-28. (Enso Bot - Mar 25, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-03-28):

**Summary:** There is **1 day** delay in implementation of the [Placement of newly opened Component Browser when nodes are selected (#180887079) task](https://www.pivotaltracker.com/story/show/180887079).
It **will cause 0 days delay** for the delivery of this weekly plan.

starting process of QA testing

**Delay Cause:** need time to implement integration tests (Enso Bot - Mar 29, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-03-28):

**Progress:** tested the PR manually, started implementation of integration tests It should be finished by 2022-03-29.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/180887079). continue implementing integration tests (Enso Bot - Mar 29, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-03-29):

**Summary:** There is **1 day** delay in implementation of the [Placement of newly opened Component Browser when nodes are selected (#180887079) task](https://www.pivotaltracker.com/story/show/180887079).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** faced another NaN issue while implementing integration tests, looks like I need to rebase onto Unit Tests branch (Enso Bot - Mar 30, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-03-29):

**Progress:** multiple meetings, investigated the compilation of unit tests with coverage on Windows. The possible fix is more difficult than I thought, so we decided to remove coverage requirement from AC of the task. Continued working on integration tests for Mateusz, faced a few issues, reported a delay. It should be finished by 2022-03-30.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/180887079). fixing integration tests (Enso Bot - Mar 30, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for today (2022-03-30):

**Progress:** Finished implementation of integration tests, faced and fixed several technical issues. Also merged Unit Tests PR. It should be finished by 2022-03-30.

**Next Day:** Next day I will be working on the [##181076594 task](https://www.pivotaltracker.com/story/show/181076594). probably solving merge conflicts in all ongoing PRs and merging them (Enso Bot - Mar 30, 2022)
<hr />
**Acceptance/QA** review set to **pass**

Tested manually and with automated integration tests (Ilya Bogdanov - Mar 31, 2022)
<hr />
**Ilya Bogdanov** reports a new **🔴 DELAY** for yesterday (2022-03-31):

**Summary:** There is **1 day** delay in implementation of the [Placement of newly opened Component Browser when nodes are selected (#180887079) task](https://www.pivotaltracker.com/story/show/180887079).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** The PR was blocked by a few other ones, namely Unit Tests PR and PR 3301 (Enso Bot - Apr 1, 2022)
<hr />
**Ilya Bogdanov** reports a new **STANDUP** for yesterday (2022-03-31):

**Progress:** rebased onto develop, tested manually and merged the PR. It should be finished by 2022-03-31. (Enso Bot - Apr 1, 2022)
<hr />
